### PR TITLE
Documented how to configure prefect server on a VM 

### PR DESCRIPTION
**Pull request by jackdesert**
## Summary

Documented how to configure prefect server on a VM 

## Changes

My first time setting up prefect server locally (on a VM because that's where I do all my development), I ran into a few hurdles. So this PR documents/clarifies the landscape for running prefect server on a VM (which is exactly like a single-node deploy, actually). 

## Importance

Helpful for those running prefect server locally who develop in a virtual machine. 

## Checklist

Documentation changes only

### CODE REVIEW

File: docs/orchestration/server/deploy-local.md [hunk 1]
```diff
@@ -31,7 +31,38 @@ You can set this directly in the UI on the Home page:
   apollo_url="http://localhost:4200/graphql"
 ```
 
-Note: The second method will change the _default_ Apollo endpoint but can still be overidden by the UI setting.
+Note: The second method will change the _default_ Apollo endpoint but can still be overridden by the UI setting.
+
+### Virtual Machine
+
+If you are running prefect server on a virtual machine, you may need to 
+configure the UI Endpoint Setting (aka apollo_url) just like you would in a server deploy.
```
Comment (cicdw) [inline docs/orchestration/server/deploy-local.md:39]:
```suggestion
configure the UI Endpoint Setting (aka `apollo_url`) just like you would in a server deploy.
```

File: docs/orchestration/server/deploy-local.md [hunk 2]
```diff
@@ -31,7 +31,38 @@ You can set this directly in the UI on the Home page:
...
+
+For example, if you access the UI at
+[http://IP_OF_VIRTUAL_MACHINE:8080](http://IP_OF_VIRTUAL_MACHINE:8080),
```
Comment (cicdw) [inline docs/orchestration/server/deploy-local.md:42]:
```suggestion
`http://IP_OF_VIRTUAL_MACHINE:8080`,
```

File: docs/orchestration/server/deploy-local.md [hunk 3]
```diff
@@ -31,7 +31,38 @@ You can set this directly in the UI on the Home page:
...
+you can open the *Network* tab of your browser's developer console and you will see
+that by default the UI makes requests to 
+[http://localhost:4200/graphql](http://localhost:4200/graphql).
```
Comment (cicdw) [inline docs/orchestration/server/deploy-local.md:45]:
```suggestion
`http://localhost:4200/graphql`
```

File: docs/orchestration/server/deploy-local.md [hunk 4]
```diff
@@ -31,7 +31,38 @@ You can set this directly in the UI on the Home page:
...
+
+If those requests are failing, simply configure the UI Endpoint Setting 
+(aka apollo_url) to point to
+[http://IP_OF_VIRTUAL_MACHINE/graphql:4200](http://IP_OF_VIRTUAL_MACHINE/graphql:4200)
```
Comment (cicdw) [inline docs/orchestration/server/deploy-local.md:49]:
```suggestion
`http://IP_OF_VIRTUAL_MACHINE/graphql:4200`
```

File: docs/orchestration/server/deploy-local.md [hunk 5]
```diff
@@ -31,7 +31,38 @@ You can set this directly in the UI on the Home page:
...
+configure the UI Endpoint Setting (aka apollo_url) just like you would in a server deploy.
+
+For example, if you access the UI at
+[http://IP_OF_VIRTUAL_MACHINE:8080](http://IP_OF_VIRTUAL_MACHINE:8080),
+you can open the *Network* tab of your browser's developer console and you will see
+that by default the UI makes requests to 
+[http://localhost:4200/graphql](http://localhost:4200/graphql).
+
+If those requests are failing, simply configure the UI Endpoint Setting 
+(aka apollo_url) to point to
```
Comment (cicdw) [inline docs/orchestration/server/deploy-local.md:48]:
```suggestion
(aka `apollo_url`) to point to
```

File: docs/orchestration/server/deploy-local.md [hunk 6]
```diff
@@ -31,7 +31,38 @@ You can set this directly in the UI on the Home page:
...
+[http://IP_OF_VIRTUAL_MACHINE/graphql:4200](http://IP_OF_VIRTUAL_MACHINE/graphql:4200)
+
+
+### Vagrant
+
+If you are a running prefect server inside a virtual machine using vagrant, 
+the easiest way to get the UI working is to simply forward ports 8080 and 4200 
+from the host to the guest.  That way the default UI Endpoint Setting 
+(aka apollo_url) will work fine.
```
Comment (cicdw) [inline docs/orchestration/server/deploy-local.md:57]:
```suggestion
(aka `apollo_url`) will work fine. 
```

**Review by cicdw — CHANGES_REQUESTED** [review]
LGTM!  I requested a few minor stylistic changes (no need to create links for the stand-in IP addresses)

**Review by madkinsz — APPROVED** [review]

File: docs/orchestration/server/deploy-local.md [hunk 7]
```diff
@@ -43,7 +43,7 @@ By default the UI will attempt to communicate with the Apollo endpoint at
 different location (e.g. if you're running Prefect Server behind a proxy), you'll need to configure the UI
 to look at a different URL.
 
-You can set this directly in the UI on the Home page:
+You can set this directly from the browser. First click *Menu* -> *Home*. In the *Connecting your Infrastructure* block, select the *Prefect Server* tab. Then you'll see this configuration option under *Connect the UI*:
```
Comment (tpdorsey) [inline docs/orchestration/server/deploy-local.md:46]:
```suggestion
You can set this directly from the browser. First click **Menu > Home**. In the **Connecting your Infrastructure** block, select the **Prefect Server** tab. Then you'll see this configuration option under **Connect the UI**: 
```

File: docs/orchestration/server/deploy-local.md [hunk 8]
```diff
@@ -55,7 +55,38 @@ You can set this directly in the UI on the Home page:
   apollo_url="http://localhost:4200/graphql"
 ```
 
-Note: The second method will change the _default_ Apollo endpoint but can still be overidden by the UI setting.
+Note: The second method will change the _default_ Apollo endpoint but can still be overridden by the UI setting.
+
+### Virtual Machine
+
+If you are running prefect server on a virtual machine, you may need to
```
Comment (tpdorsey) [inline docs/orchestration/server/deploy-local.md:62]:
```suggestion
If you are running prefect server on a virtual machine (VM), you may need to 
```

File: docs/orchestration/server/deploy-local.md [hunk 9]
```diff
@@ -55,7 +55,38 @@ You can set this directly in the UI on the Home page:
   apollo_url="http://localhost:4200/graphql"
 ```
 
-Note: The second method will change the _default_ Apollo endpoint but can still be overidden by the UI setting.
+Note: The second method will change the _default_ Apollo endpoint but can still be overridden by the UI setting.
+
+### Virtual Machine
+
+If you are running prefect server on a virtual machine, you may need to 
+configure the UI Endpoint Setting (aka `apollo_url`) just like you would in a server deploy.
```
Comment (tpdorsey) [inline docs/orchestration/server/deploy-local.md:63]:
```suggestion
configure the UI Endpoint Setting (`apollo_url`) just like you would in a server deploy.
```

File: docs/orchestration/server/deploy-local.md (hunk 8 above)
Comment (tpdorsey) [inline docs/orchestration/server/deploy-local.md:62]:
```suggestion
If you are running Prefect server on a virtual machine, you may need to 
```

File: docs/orchestration/server/deploy-local.md [hunk 10]
```diff
@@ -55,7 +55,38 @@ You can set this directly in the UI on the Home page:
...
+
+For example, if you access the UI at
+`http://IP_OF_VIRTUAL_MACHINE:8080`,
+you can open the *Network* tab of your browser's developer console and you will see
+that by default the UI makes requests to 
+`http://localhost:4200/graphql`
+
+If those requests are failing, simply configure the UI Endpoint Setting 
+(aka `apollo_url`) to point to
```
Comment (tpdorsey) [inline docs/orchestration/server/deploy-local.md:72]:
```suggestion
(`apollo_url`) to point to
```

File: docs/orchestration/server/deploy-local.md [hunk 11]
```diff
@@ -55,7 +55,38 @@ You can set this directly in the UI on the Home page:
...
+`http://IP_OF_VIRTUAL_MACHINE/graphql:4200`
+
+
+### Vagrant
+
+If you are a running prefect server inside a virtual machine using vagrant,
```
Comment (tpdorsey) [inline docs/orchestration/server/deploy-local.md:78]:
```suggestion
If you are a running Prefect server inside a VM using vagrant, 
```

File: docs/orchestration/server/deploy-local.md [hunk 12]
```diff
@@ -55,7 +55,38 @@ You can set this directly in the UI on the Home page:
...
+(aka `apollo_url`) to point to
+`http://IP_OF_VIRTUAL_MACHINE/graphql:4200`
+
+
+### Vagrant
+
+If you are a running prefect server inside a virtual machine using vagrant, 
+the easiest way to get the UI working is to simply forward ports 8080 and 4200 
+from the host to the guest.  That way the default UI Endpoint Setting 
+(aka `apollo_url`) will work fine.
```
Comment (tpdorsey) [inline docs/orchestration/server/deploy-local.md:81]:
```suggestion
(`apollo_url`) will work fine. 
```

File: docs/orchestration/server/deploy-local.md (hunk 11 above)
Comment (tpdorsey) [inline docs/orchestration/server/deploy-local.md:78]:
```suggestion
If you are a running prefect server inside a virtual machine using Vagrant, 
```

**Review by tpdorsey — APPROVED** [review]

**Comment by tpdorsey** [comment]
Looks good. Thank you!

**Review by madkinsz — APPROVED** [review]

**Pull request merged**